PYME/cluster/clusterUI/recipes/views.py

```diff
@@ -28,31 +29,30 @@ def get_input_glob(request):
 
 def run(request):
     from PYME import config
-    if config.get('PYMERuleserver-use', True):
-        from PYME.cluster.HTTPRulePusher import RecipePusher
-    else:
-        from PYME.cluster.HTTPTaskPusher import RecipePusher
-    recipeURI = ('pyme-cluster://%s/' % server_filter) + request.POST.get('recipeURL').lstrip('/')
-
-    pusher = RecipePusher(recipeURI=recipeURI)
-
-
-    fileNames = request.POST.getlist('files', [])
-    pusher.fileTasksForInputs(input=fileNames)
+    from PYME.cluster.rules import RecipeRule
+    import posixpath
 
+    recipeURI = ('pyme-cluster://%s/' % server_filter) + request.POST.get('recipeURL').lstrip('/')
+    input_list = request.POST.getlist('files', [])
+    try:
+        # put outputs in the same directory as selected files, under 'analysis'
```

Comment:
I don't like this as it has the potential to get super messy if you are analysing files across directories. Approach to date was to hard-code the output directory in the recipe / not rely on substitution (this is safe, if gross), a better option might be to have it as a UI input as in `run_template()`.

Reply:
how do you analyze files across directories in this view?

Reply:
agree ui input would be best

Reply:
This is used by `recipe_standalone()` which accepts globs for inputs.

Reply:
as well as the lightweight form in the files view.

Reply:
Pre-populating the output directory as current directory + /analysis might not be a bad idea, but probably better done in the html view.